v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/templates/SpannerChangeStreamsToElasticsearch.java

```diff
@@ -0,0 +1,175 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.elasticsearch.templates;
+
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.Options.RpcPriority;
+import com.google.cloud.teleport.v2.elasticsearch.options.SpannerChangeStreamsToElasticsearchOptions;
+import com.google.cloud.teleport.v2.elasticsearch.transforms.WriteToElasticsearch;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.beam.runners.dataflow.options.DataflowPipelineWorkerPoolOptions;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.io.gcp.spanner.SpannerConfig;
+import org.apache.beam.sdk.io.gcp.spanner.SpannerIO;
+import org.apache.beam.sdk.io.gcp.spanner.changestreams.model.ColumnType;
+import org.apache.beam.sdk.io.gcp.spanner.changestreams.model.DataChangeRecord;
+import org.apache.beam.sdk.io.gcp.spanner.changestreams.model.ModType;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SpannerChangeStreamsToElasticsearch} pipeline forwards Spanner Change stream data to
+ * Elasticsearch.
+ */
+public class SpannerChangeStreamsToElasticsearch {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(SpannerChangeStreamsToElasticsearch.class);
+  private static final String USE_RUNNER_V2_EXPERIMENT = "use_runner_v2";
+
+  public static void main(String[] args) {
+    LOG.info("Starting Input Files to GCS");
+
+    SpannerChangeStreamsToElasticsearchOptions options =
+        PipelineOptionsFactory.fromArgs(args).as(SpannerChangeStreamsToElasticsearchOptions.class);
+
+    run(options);
+  }
+
+  private static String getProjectId(SpannerChangeStreamsToElasticsearchOptions options) {
+    return options.getSpannerProjectId().isEmpty()
+        ? options.getProject()
+        : options.getSpannerProjectId();
+  }
+
+  public static PipelineResult run(SpannerChangeStreamsToElasticsearchOptions options) {
+    options.setStreaming(true);
+    options.setEnableStreamingEngine(true);
+    options.setAutoscalingAlgorithm(
+        DataflowPipelineWorkerPoolOptions.AutoscalingAlgorithmType.NONE);
+    // Spanner change stream updates would require partial updates when writing to Elasticsearch
```

Comment:
Is this something specific to the ElasticSearch template? What effect would this have?